### ISSUE TITLE
TranslatableComponents aren't translated when sending it to player using ServerPlayer#sendMessage

### ISSUE THREAD

**Issue by CDFN** [SpongePowered/Sponge#3326]
**I am currently running**

- SpongeForge version: -
- SpongeVanilla version: 1.16.5-8.0.0
- Java version: 15
- Operating System: Windows 10

- Plugins/Mods: Only test plugin in which issue occurrs

**Issue Description**
ServerPlayer#sendMessage doesn't translate TranslatableComponents

```java
    var toTranslate = Component.translatable("hub.someText");
    event.getPlayer().sendMessage(toTranslate);
    logger.info(GlobalTranslator.render(toTranslate, Locales.EN_US));
```

produces:
![image](https://user-images.githubusercontent.com/41289688/110683149-044ee680-81dc-11eb-8c20-adc6831d1ce3.png)
`[20:02:28] [Server thread/INFO] [cdgames_hub]: TextComponentImpl{content="epicTextContent", style=StyleImpl{color=null, obfuscated=not_set, bold=not_set, strikethrough=not_set, underlined=not_set, italic=not_set, clickEvent=null, hoverEvent=null, insertion=null, font=null}, children=[]}`

Message sent to player isn't translated, message rendered using GlobalTranslator is translated.

**Comment by kashike** [comment]
Show all your code, including the registration and loading of your translator.

**Comment by CDFN** [comment]
Oh, should've done that when submitting issue, apologies.

![image](https://user-images.githubusercontent.com/41289688/110685429-8cce8680-81de-11eb-9597-e92f0f98815b.png)
```java
  @Listener
  public void onServerStart(LoadedGameEvent event) {
    logger.log(Level.INFO, "Loading {}", pluginContainer.getMetadata().getId());

    var registry = TranslationRegistry.create(Key.key("hub", "hub_translation_registry"));
    registry.registerAll(Locales.EN_US, PropertyResourceBundle.getBundle("lang/lang", Locales.EN_US), true);
    GlobalTranslator.get().addSource(registry);

    var serverSideJoinListener = injector.getInstance(ServerSideJoinListener.class);
    eventManager.registerListeners(pluginContainer, serverSideJoinListener);
  }
```

```java
  @Inject
  public ServerSideJoinListener(Logger logger) {
    this.logger = logger;
  }

  @Listener
  public void onPlayerJoin(ServerSideConnectionEvent.Join event) {
    event.getPlayer().getInventory().clear();
    event.getPlayer().getInventory().getHotbar().set(0, COMPASS_ITEM);

    logger.info(event.getPlayer().getLocale().toString());

    var toTranslate = Component.translatable("hub.someText");
    event.getPlayer().sendMessage(toTranslate);
    logger.info(GlobalTranslator.render(toTranslate, Locales.EN_US));
  }
````

**Comment by zml2008** [comment]
fixed in 9b93dd8b1ef921daecfd6cec092afbe65e6f6a34
